src/wasm-binary.h

```diff
@@ -1195,7 +1195,7 @@ class WasmBinaryBuilder {
   bool debug;
 
   size_t pos = 0;
-  int32_t startIndex = -1;
+  int64_t startIndex = -1;
```

Comment:
Yes, this should be `uint32_t`, unsigned. But really, it should be the `Index` type.

(In theory it's also bad that we use `-1` as an invalid value, but the possibility of all 32 bits being used as function indexes is not practical. Might be worth a TODO note though.)